weblate/api/views.py

```diff
@@ -339,9 +339,196 @@ def destroy(self, request, *args, **kwargs):
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        obj.languages.remove(language)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @action(
+        detail=True, methods=["post"],
+    )
+    def projects(self, request, **kwargs):
+        obj = self.get_object()
+        self.perm_check(request)
+
+        if "project_id" not in request.data:
+            raise ParseError("Missing project_id parameter")
+
+        try:
+            project = Project.objects.get(pk=int(request.data["project_id"]),)
+        except (Project.DoesNotExist, ValueError) as error:
+            return Response(
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        obj.projects.add(project)
+        serializer = self.serializer_class(obj, context={"request": request})
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    @action(detail=True, methods=["delete"], url_path="projects/(?P<project_id>[^/.]+)")
+    def delete_projects(self, request, id, project_id):
+        obj = self.get_object()
+        self.perm_check(request)
+
+        try:
+            project = Project.objects.get(pk=int(project_id))
+        except (Project.DoesNotExist, ValueError) as error:
+            return Response(
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        obj.projects.remove(project)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @action(
+        detail=True, methods=["post"],
+    )
+    def componentlist(self, request, **kwargs):
+        obj = self.get_object()
+        self.perm_check(request)
+
+        if "component_list_id" not in request.data:
+            raise ParseError("Missing component_list_id parameter")
+
+        try:
+            component_list = ComponentList.objects.get(
+                pk=int(request.data["component_list_id"]),
+            )
+        except (ComponentList.DoesNotExist, ValueError) as error:
+            return Response(
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        obj.componentlist = component_list
+        obj.save()
+        serializer = self.serializer_class(obj, context={"request": request})
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    @action(
+        detail=True,
+        methods=["delete"],
+        url_path="componentlist/(?P<component_list_id>[^/.]+)",
+    )
+    def delete_componentlist(self, request, id, component_list_id):
+        obj = self.get_object()
+        self.perm_check(request)
+        try:
+            ComponentList.objects.get(
+                pk=int(component_list_id),
+            )
```

Comment:
```suggestion
            ComponentList.objects.get(pk=int(component_list_id))
```

I suggest you to install pre-commit, it will do the formatting automatically before git commit, see https://docs.weblate.org/en/latest/contributing.html#coding-standard

Reply:
On it.